### PR TITLE
Use Github actions to run CI tests

### PR DESCRIPTION
**Pull request by MonkeyDo**
Travis is dead, long live CI !

Travis CI stopped their OSS-friendly programs, and as a consequence we can't use it for testing anymore.

Instead, we're moving testing to Github actions, like we have been doing successfully with [ListenBrainz](https://github.com/metabrainz/listenbrainz-server/tree/master/.github/workflows)

### CODE REVIEW

File: .github/workflows/test-report.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Test Report
+on:
+  workflow_run:
+    workflows: ['CI']                     # runs after CI workflow
+    types:
+      - completed
+jobs:
```
Comment (amCap1712) [inline .github/workflows/test-report.yml:7]: Why not just run this as an extra step after exporting test results?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      # For more information, see https://docs.npmjs.com/cli/ci.html
+      - name: Install dependencies
+        run: npm ci
+      
+      - name: Run tests
+        run: npm run test-ci                    # run tests (configured to use mocha's json reporter)
+      
+      - name: Export test results
+        uses: actions/upload-artifact@v2  # upload test results
+        if: success() || failure()        # run this step even if previous step failed
```
Comment (amCap1712) [inline .github/workflows/ci.yml:77]: There's always too ;) https://docs.github.com/en/actions/learn-github-actions/expressions#always

File: .github/workflows/test-report.yml (hunk 1 above)
Comment (MonkeyDo) [reply .github/workflows/test-report.yml:7]: Following the instructions on the test-reporter action:
https://github.com/dorny/test-reporter#recommended-setup-for-public-repositories

I haven't tested it for real yet, so not 100% sure it's necessary, but it certainly won't hurt to try as advertised first.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (MonkeyDo) [reply .github/workflows/ci.yml:77]: I suppose that doesn't include the `cancelled()` status, which `always()` does include according to the docs. Probably best not to run if the job is cancelled.

File: .github/workflows/test-report.yml (hunk 1 above)
Comment (MonkeyDo) [reply .github/workflows/test-report.yml:7]: I'll keep this comment unresolved and remember to try it out.

**Review by amCap1712 — APPROVED** [review]

File: .github/workflows/test-report.yml (hunk 1 above)
Comment (MonkeyDo) [reply .github/workflows/test-report.yml:7]: Re-read this article and confirming the recommended "safe" way to check out PRs from forked repos without allowing the forked repo to have write access to our repo is to run any task that requires write access as a separate workflow where the context will be safe

https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

**Pull request merged**